http.go

```diff
@@ -56,6 +56,9 @@ type Request struct {
 	// Request timeout. Usually set by DoDeadline or DoTimeout
 	// if <= 0, means not set
 	timeout time.Duration
+
+	// Allow to change Host header with parsedURI == true (E.g. Client for type).
+	AllowToChangeHostHeader bool
```

Comment:
I think something like this is cleaner?
```suggestion
	// Use Host header (request.Header.SetHost) instead of the host from SetRequestURI, SetHost, or URI().SetHost
	UseHostHeader bool
```

Reply:
Good comment. I've rename attribute and add test for it.